core/src/test/java/org/elasticsearch/index/query/TemplateQueryBuilderTests.java

```diff
@@ -80,7 +80,8 @@ protected void assertBoost(TemplateQueryBuilder queryBuilder, Query query) throw
     @Override
     public void testUnknownField() throws IOException {
         TemplateQueryBuilder testQuery = createTestQueryBuilder();
-        String queryAsString = testQuery.toString().replace("inline", "bogusField");
+        String testQueryAsString = toXContent(testQuery, randomFrom(XContentType.JSON, XContentType.YAML)).string();
```

Comment:
yes we always print the query as a string using json. That said toXContent supports different formats, as well as elasticsearch. I think it is good testing with different formats. It would be a bit more complicated in the base class, cause string replacement would need to be different depending on whether we are using yaml or json, but this one was easy. Note that when we do string replacement the binary formats are not used...